### PR TITLE
[bitnami/fluentd]

### PR DESCRIPTION
**Pull request by Gwenael22**
**Description of the change**
fix TLS documentation and add root CA certificate as documented

**Benefits**
more comprehensive configuration and possibility to add your own certificate CA certificate

**Possible drawbacks**

**Applicable issues**

  - fixes #

**Additional information**

**Checklist** 

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/fluentd/templates/tls-certs.yaml [hunk 1]
```diff
@@ -16,6 +16,7 @@ type: kubernetes.io/tls
 data:
   tls.crt: {{ .certificate | b64enc }}
   tls.key: {{ .key | b64enc }}
+  rootCA.crt: {{ .caCertificate | b64enc }}
```
Comment (carrodher) [inline bitnami/fluentd/templates/tls-certs.yaml:19]: `tls.caCertificate` is not present in the values.yaml, see https://github.com/bitnami/charts/commit/83ad4ad8664d6a975beca04ed90800a4efd3a40c. It should be removed from the README table and not used here

File: bitnami/fluentd/README.md [hunk 2]
```diff
@@ -216,8 +216,8 @@ The following tables lists the configurable parameters of the fluentd chart and
 | `metrics.serviceMonitor.annotations`            | ServiceMonitor annotations                                                                                     | `nil`                                                                                                   |
 | `tls.enabled`                                   | Enable the addition of TLS certificates                                                                        | `false`                                                                                                 |
 | `tls.caCertificate`                             | Ca certificate                                                                                                 | Certificate Authority (CA) bundle content                                                               |
```
Comment (carrodher) [inline bitnami/fluentd/README.md:218]: This parameter doesn't exist

**Review by carrodher — COMMENTED** [review]
Thanks for your contribution, please take a look at the above comments

File: bitnami/fluentd/templates/tls-certs.yaml (hunk 1 above)
Comment (Gwenael22) [reply bitnami/fluentd/templates/tls-certs.yaml:19]: Ok, the tls.caCertificate was removed but how can i configure a custom certificate autority ?
I need to use this type of following configuration but i could not declare CA certicate for the moment : 
```
<transport tls>
            cert_path /opt/bitnami/fluentd/certs/tls.crt
            private_key_path /opt/bitnami/fluentd/certs/tls.key
            ca_path /opt/bitnami/fluentd/certs/rootCA.crt
</transport>
````

File: bitnami/fluentd/README.md (hunk 2 above)
Comment (carrodher) [reply bitnami/fluentd/README.md:218]: It seems it was removed by mistake, so it is fine maintaining it in the README, we should add it to the values

File: bitnami/fluentd/templates/tls-certs.yaml (hunk 1 above)
Comment (carrodher) [reply bitnami/fluentd/templates/tls-certs.yaml:19]: Yes, it is true, in that case. Can you add the parameter in the values.yaml? In that case the certificate should be mounted as a volume, see other examples like `phpmyadmin/templates/deployment.yaml`

**Review by carrodher — COMMENTED** [review]
Thanks, caCertificate parameter is missing in the values as well as the logic to mount it as a volume. Can you include those changes in the PR to make it fully functional?

File: bitnami/fluentd/templates/tls-certs.yaml (hunk 1 above)
Comment (carrodher) [reply bitnami/fluentd/templates/tls-certs.yaml:19]: In this file, can you add a similar entry to the ones regarding `tls.crt` and `tls.key` at the end of the file?
```diff
 data:
   tls.crt: {{ required "A valid .Values.tls.certificate entry required!" .Values.tls.certificate | b64enc }}
   tls.key: {{ required "A valid .Values.tls.key entry required!" .Values.tls.key | b64enc }}
+  rootCA.crt: {{ required "A valid .Values.tls.caCertificate entry required!" .Values.tls.caCertificate | b64enc }}
```

File: bitnami/fluentd/templates/aggregator-statefulset.yaml [hunk 3]
```diff
@@ -146,6 +146,8 @@ spec:
                 path: tls.crt
               - key: tls.key
                 path: tls.key
+              - key: rootCA.crt
```
Comment (carrodher) [inline bitnami/fluentd/templates/aggregator-statefulset.yaml:149]: Can you apply the same change in the `forwarder-daemonset.yaml` file?

**Review by carrodher — CHANGES_REQUESTED** [review]
Thanks for applying some changes! Apart from the inline comments, it is also needed to add the `caCertificates` parameters to the _values.yaml_:
```yaml
 ## Mount TLS certificates
 ##
 tls:
   enabled: false
   certificate: |-
   key: |-
   caCertificate: |-
   # existingSecret: name-of-existing-secret-to-certificates
```

**Review by carrodher — APPROVED** [review]
Thanks for your changes! The PR is going to be merged, but then some internal steps are needed before release the chart in the Helm Bitnami repo, see https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#pr-approval-and-release-process

**Comment by Gwenael22** [comment]
thank you @carrodher

**Pull request merged**